src/compiler/watch.ts

```diff
@@ -721,10 +744,10 @@ namespace ts {
             // there was version update and new source file was created.
             if (hostSourceFileInfo) {
                 // record the missing file paths so they can be removed later if watchers arent tracking them
-                if (isString(hostSourceFileInfo)) {
+                if (isFileMissingOnHost(hostSourceFileInfo)) {

```

Comment:
In the line below the hostSourceFile could be ```FilePresenceUnknownOnHost```  or ```FileMayBePresentOnHost``` . It is kind of redundant to check if ```isFilePresentOnHost``` since it only checks if sourceFile is present on the host and the next check is going to check if sourceFile === oldSourceFile (oldSourceFile is never undefined). Hence the cast